### PR TITLE
feat(pkg/diffusers): add basic project structure, abstract mixin and pipeline of sdxl

### PR DESCRIPTION
**Pull request by geniuspatrick**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Comment by SamitHuang** [comment]
A document is needed to introduce the new APIs, like how to config, how to run text-to-image generation in diffusers-like style

**Comment by SamitHuang** [comment]
TODO: 
1.  docs
2. mixed precision setting, data sink, and other additional settings for MindSpore. wrap in context manageer?
3. correct inference results and examples
4. correct training results

File: examples/diffusers/text_to_image/train_text_to_image_sdxl.py [hunk 1]
```diff
@@ -0,0 +1,1095 @@
...
+        variant=args.variant,
+    )
+    unet = UNet2DConditionModel.from_pretrained(
+        args.pretrained_model_name_or_path, subfolder="unet", revision=args.revision, variant=args.variant
+    )
+
+    # Freeze vae and text encoders.
+    def freeze_params(m: nn.Cell):
+        for p in m.get_parameters():
+            p.require_grad = False
```
Comment (CaitinZhao) [inline examples/diffusers/text_to_image/train_text_to_image_sdxl.py:584]: require_grad=False will not cut off the gradient calculate, it's better to use stop_gredient?

File: examples/diffusers/text_to_image/train_text_to_image_sdxl.py [hunk 2]
```diff
@@ -0,0 +1,1095 @@
...
+            # with accelerator.accumulate(unet): todo: support accumulation
+            loss = train_step(model_input, prompt_embeds, pooled_prompt_embeds, add_time_ids)
+
+            progress_bar.update(1)
+            global_step += 1
+
+            if is_master(args):
+                if global_step % args.checkpointing_steps == 0:
+                    # _before_ saving state, check if this save would set us over the `checkpoints_total_limit`
+                    if args.checkpoints_total_limit is not None:
```
Comment (CaitinZhao) [inline examples/diffusers/text_to_image/train_text_to_image_sdxl.py:885]: extract the checkpoint module?

File: mindone/diffusers/models/activations.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mindspore as ms
+from mindspore import nn, ops
+
+
+def sigmoid(x):
+    """A numerically stable version of the logistic sigmoid function."""
+    return ops.where(
```
Comment (CaitinZhao) [inline mindone/diffusers/models/activations.py:22]: 这个为什么单独实现？

File: mindone/diffusers/models/activations.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        x >= 0.0,
+        1.0 / (1.0 + ops.exp(-x)),  # For positive values
+        ops.exp(x) / (1.0 + ops.exp(x))  # For negative values
+    )
+
+
+class SiLU(nn.Cell):
```
Comment (CaitinZhao) [inline mindone/diffusers/models/activations.py:29]: 这个也是

File: mindone/diffusers/models/normalization.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+import numbers
+from typing import Dict, Optional, Tuple
+
+import mindspore as ms
+from mindspore import nn, ops, Tensor, Parameter
+from mindspore.common.initializer import initializer
+
+
+class LayerNorm(nn.Cell):
```
Comment (CaitinZhao) [inline mindone/diffusers/models/normalization.py:24]: 这个为什么要自己写而不是用mindspore的接口呢

File: mindone/diffusers/optimization.py [hunk 6]
```diff
@@ -0,0 +1,348 @@
+# coding=utf-8
```
Comment (CaitinZhao) [inline mindone/diffusers/optimization.py:1]: 这块是可以用这口改造，完全对标torch做的，https://www.mindspore.cn/docs/zh-CN/r2.2/api_python/mindspore.experimental.html

File: mindone/diffusers/models/activations.py (hunk 3 above)
Comment (geniuspatrick) [reply mindone/diffusers/models/activations.py:22]: 这会有更好的精度

File: mindone/diffusers/models/normalization.py (hunk 5 above)
Comment (geniuspatrick) [reply mindone/diffusers/models/normalization.py:24]: 1. 默认值
2. 参数名称
3. GN仅支持4维，扩展之

File: mindone/diffusers/optimization.py (hunk 6 above)
Comment (geniuspatrick) [reply mindone/diffusers/optimization.py:1]: cool，我会将其列为todo，至正式支持

**Review by CaitinZhao — APPROVED** [review]

**Review by vigo999 — APPROVED** [review]
